pkg/cmd/util/clientcmd/factory.go

```diff
@@ -277,11 +281,18 @@ func (c *clientCache) ClientConfigForVersion(version string) (*kclient.Config, e
 		c.defaultConfig = config
 	}
 	// TODO: have a better config copy method
+	if config, ok := c.configs[version]; ok {
+		return config, nil
+	}
+	// TODO: have a better config copy method
 	config := *c.defaultConfig
-	if len(version) != 0 {
-		config.Version = version
+	negotiatedVersion, err := kclient.NegotiateVersion(nil, &config, version, latest.Versions)
```

Comment:
> It's using latest.Versions from "github.com/openshift/origin/pkg/api/latest"

Take a look here: https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/helper.go#L201 and then follow the chain down to here https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/client.go#L150, which ends up using using the `baseURL` set in `New` with `SetKubernetesDefault` here https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/helper.go#L290.

That's all just a rough inspection.  `--loglevel=8` would be authoritative.